packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -723,8 +723,6 @@ class BuildOwner {
     });
     try {
       callback();
-    } catch (e, stack) {
-      _debugReportException(context, e, stack);
     } finally {
       assert(() {
         _debugStateLockLevel -= 1;
```

Comment:
we should probably check at some point whether the compiler is clever enough to elide the exception handling for finally blocks that end up empty